src/blinding.cpp

```diff
@@ -298,6 +301,37 @@ std::pair<uc32, cleared_uc32> blind25_key_pair(
     return result;
 }
 
+static const auto version_blinding_hash_key_sig = to_unsigned_sv("VersionCheckKey_sig"sv);
+
+std::pair<uc32, cleared_uc64> blind_version_key_pair(ustring_view ed25519_sk) {
+    std::array<unsigned char, 64> ed_sk_tmp;
+    if (ed25519_sk.size() == 32) {
+        std::array<unsigned char, 32> pk_ignore;
+        crypto_sign_ed25519_seed_keypair(pk_ignore.data(), ed_sk_tmp.data(), ed25519_sk.data());
+        ed25519_sk = {ed_sk_tmp.data(), 64};
+    }
```

Comment:
This temp sk ought to go into a cleared_uc64, if needed:
```suggestion
    std::optional<cleared_uc64> ed_sk_tmp;
    if (ed25519_sk.size() == 32) {
        std::array<unsigned char, 32> pk_ignore;
        ed_sk_tmp.emplace();
        crypto_sign_ed25519_seed_keypair(pk_ignore.data(), ed_sk_tmp->data(), ed25519_sk.data());
        ed25519_sk = {ed_sk_tmp->data(), 64};
    }
```

Reply:
Actually I don't think this get-the-full-secret-key code is needed at all for this function.